### PR TITLE
Added interface to the register

### PR DESCRIPTION
**Pull request by jayashreekumaresan2026**
Fixes https://github.com/Quant-Team/qvm/issues/10

### CODE REVIEW

**Comment by sah4ez** [comment]
Thanks for your PR, it's great that project interesting somebody)

File: pkg/machine/register.go [hunk 1]
```diff
@@ -22,27 +27,27 @@ var (
...
-func (r *Register) Measure() []circuit.Qubiter {
+func (r *register) Measure() []circuit.Qubiter {
 	for i := range r.q {
 		r.q[i] = r.q[i].Measure()
 	}
 	return r.q
 }
 
-func NewRegister(cfg *config.Register, state []Bit) (r *Register, err error) {
+func NewRegister(cfg *config.Register, state []Bit) (Register, error) {
```
[the remark this answers is not in the document]
Comment (jayashreekumaresan2026) [reply pkg/machine/register.go]: Fixed the PR as per your suggestion.

File: pkg/machine/register.go [hunk 2]
```diff
@@ -7,7 +7,12 @@ import (
 	"github.com/Quant-Team/qvm/pkg/config"
 )
 
-type Register struct {
```
[the remark this answers is not in the document]
Comment (sah4ez) [reply pkg/machine/register.go]: I will do it myself

**Comment by sah4ez** [comment]
@jayashreekumaresan2026 thx! :tada:

**Pull request merged**